### ISSUE TITLE
What's the purpose of lastTimingPoint in DSP?

### ISSUE THREAD

**Issue by TsFreddie** [Drewol/unnamed-sdvx-clone#184]
TLDR; The `lastTimingPoint` member in DSP appears to be making the Retrigger effect sound wrong. 
Related issues: #34, #15.

---

**More Info:**

There is a `lastTimingPoint` as a member in DSP, which appears to be used only in RetriggerDSP:
https://github.com/Drewol/unnamed-sdvx-clone/blob/e2a331a67d25b8749e4a8a50e5d2cf175a8af6e7/Audio/src/DSP.cpp#L302-L303

By the look of it, it determine the start position of the repeating sample.
But it is set to current playback time in `AudioPlayback::SetEffect`,
https://github.com/Drewol/unnamed-sdvx-clone/blob/a0f8a90c81c1552a9146faeb533676172c40f8f9/Main/Audio/AudioPlayback.cpp#L197

which is (100ms?) early and does not take audio offset into account.
See the code in `BeatmapPlayback::Update`:
https://github.com/Drewol/unnamed-sdvx-clone/blob/e4826ebc635e7080f508128ea2cc4099f25afdaa/Beatmap/src/BeatmapPlayback.cpp#L194-L202

I tried to set lastTimingPoint to `object->time` in `AudioPlayback::SetEffect` which, in result, sound a lot better on my problematic map (with 0 offset).

I need more context about this `lastTimingPoint` variable before I jumped to the conclusion that it is useless.

**Comment by TsFreddie** [comment]
Also the `chartOffset` member of DSP doesn't seems to be used anywhere at all. Both of these looks like a problem for the Retrigger effect.
Nonetheless, I tested on a map with 0 offset, which still sound wrong compared to KSM.

**Comment by TsFreddie** [comment]
Upon further inspection, it appears that BPM change behave differently in KSM and USC. In KSM, BPM change doesn't reset to first bar, where as the USC does reset. (Hence a different starting point for repeating pattern in Retrigger in USC)

![image](https://user-images.githubusercontent.com/3797859/58367923-d53c8780-7f17-11e9-8342-b2e96f11d396.png)

I don't think change a mechanic is desperately needed, since I can just change my map to reset the bar line manually myself.

However, I did find a precision issue in timing that cause some Retrigger effect sounded wrong, but it is not related to this issue, I'll close this issue for now.
